desk/tests/app/reel.hoon

```diff
@@ -0,0 +1,323 @@
+/-  reel, gv=groups-ver, meta
+/+  *test-agent, test, s=subscriber, t=contacts
+/=  reel-agent  /app/reel
+|%
++$  state-5
+  $:  %5
+      vic=@t
+      civ=ship
+      our-profile=contact:t
+      our-metadata=(map token:reel metadata:reel)
+      open-link-requests=(set (pair ship cord))
+      open-describes=(set token:reel)
+      stable-id=(map cord token:reel)
+      =^subs:s
+  ==
+++  dap  %reel-test
+++  provider  ~loshut-lonreg
+++  group-invite-meta
+  ^~
+  ^-  metadata:reel
+  :-  %group-0
+  %-  my
+  :~  ['inviterUserId' '~sampel-palnet']
+      ['inviterNickname' 'Sampel Palnet']
+      ['inviterAvatarImage' 'https://sampel-palnet.arvo.network/avatar.png']
+      ['invitedGroupTitle' 'Sunrise']
+      ['invitedGroupDescription' '']
+      ['invitedGroupId' '~sampel-palnet/sunrise']
+      ['invitedGroupIconImageUrl' 'https://sampel-palnet.arvo.network/sunrise.jpg']
+      ['bite-type' '2']
+  ==
+++  personal-invite-meta
+  ^~
+  ^-  metadata:reel
+  :-  %group-0
+  %-  my
+  :~  ['inviterUserId' '~sampel-palnet']
+      ['inviterNickname' 'Sampel Palnet']
+      ['inviterAvatarImage' 'https://sampel-palnet.arvo.network/avatar.png']
+      ['inviteType' 'user']
+      ['invitedGroupId' '~zod/personal-invite-link']
+      ['bite-type' '2']
+  ==
+++  do-register-invite
+  |=  [=token:reel =metadata:reel]
+  =/  m  (mare ,(list card))
+  ^-  form:m
+  =+  nonce=(scot %da ~2025.9.3)
+  ;<  ~  bind:m  (jab-bowl |=(=bowl bowl(now ~2025.9.3)))
+  ;<  ~  bind:m   (set-src ~sampel-palnet)
+  =+  id=(~(got by fields.metadata) 'invitedGroupId')
+  ;<  caz=(list card)  bind:m  (do-poke reel-describe+!>([id metadata]))
+  ;<  ~  bind:m
+    %+  ex-cards  caz
+    :~  (ex-poke /describe [provider %bait] bait-describe+!>([nonce metadata]))
+    ==
+  ;<  *  bind:m  (set-src provider)
+  ;<  caz=(list card)  bind:m  (do-poke reel-confirmation+!>([nonce token]))
+  (pure:m caz)
+++  get-full-peek
+  |*  [=mold =path]
+  =/  m  (mare mold)
+  ^-  form:m
+  |=  =state
+  =/  res  ((get-peek path) state)
+  ?:  ?=(%| -.res)  res
+  =/  peek=(unit (unit cage))
+    +<.res
+  ?~  peek
+    |+~['invalid scry path' (spat path)]
+  ?~  u.peek
+    |+~['unexpected empty result at scry path' (spat path)]
+  ::XX there is a compiler bug here if .q face is missing
+  ::   from the peek. it compiles, but fails at runtime when
+  ::   the result is dereferenced.
+  ::
+  &+[!<(mold q.u.u.peek) state]
```

Comment:
You mean, on the cage as produced by the agent? Weird either way, even if that doesn't cast, because the `=/  peek` above casts and as such should have the faces on there always...

Reply:
I think the problem was with
`=/  peek  +<.res`
which does not seem to result in a face, even though the type should be there, no?
Above would give a compiler error though – unfortunately I did not preserve a version which seemed to fail at compile time.

`=/  peek  out.p.res`, on the other hand, will work properly.